scripts/build_layers.sh

```diff
@@ -29,19 +29,23 @@ function make_path_absolute {
 }
 
 function docker_build_zip {
-    # Args: [ruby version] [zip destination]
+    # Args: [ruby version] [zip destination] [architecture]
 
     destination=$(make_path_absolute $2)
+    arch=$3
 
     # Install datadog ruby in a docker container to avoid the mess from switching
     # between different ruby runtimes.
     temp_dir=$(mktemp -d)
-    docker build -t datadog-lambda-layer-ruby:$1 . --no-cache \
-        --build-arg "image=lambci/lambda:build-ruby${1}" --build-arg "runtime=${1}.0"
+    docker buildx build -t datadog-lambda-ruby-${arch}:$1 . --no-cache \
+        --build-arg "image=ruby:${1}" \
+        --build-arg "runtime=${1}.0" \
+        --platform linux/${arch} \
+        --load
 
     # Run the image by runtime tag, tar its generatd `ruby` directory to sdout,
     # then extract it to a temp directory.
-    docker run --rm datadog-lambda-layer-ruby:$1 tar cf - /opt/ruby | tar -xf - -C $temp_dir
+    docker run --rm datadog-lambda-ruby-${arch}:$1 tar cf - /opt/ruby | tar -xf - -C $temp_dir
```

Comment:
Do we only ship `/opt/ruby` to the client? Nothing else is needed for the layer? @duncanista

Reply:
That is correct, since this layer is later merged in the AWS Lambda runtime `/opt/ruby`, this is enough to make it work!